yellowbrick/features/base.py

```diff
@@ -20,110 +20,184 @@
+            Dependent target dataset optionally used for training.
 
-        Fits the visualizer to X and y with opetional parameters by passing in
-        all of kwargs, then calls poof with the same kwargs. This method must
-        return the result of the transform method.
+        kwargs : dict, optional
+            Keyword arguments to pass to the ``poof()`` method.
+
+        Returns
+        -------
+        Xp : array-like, shape (m_samples, m_features)
+            The transformed dataset X prime.
         """
-        Xp = self.fit_transform(X, y, **kwargs)
+        Xp = self.fit_transform(X, y)
         self.poof(**kwargs)
         return Xp
 
 
 class MultiFeatureVisualizer(FeatureVisualizer):
-    """
-    MultiFeatureVisualiers are a subclass of FeatureVisualizer that visualize
-    several features at once. This class provides base functionality for
-    getting the names of features for use in plot annotation.
+    """Direct visualization of a feature set.
+
+    MultiFeatureVisualiers visualize several features at once, usually in order
+    to compare the effectiveness of a subset of features to the superset. This
+    type of visualizer provides base functionality for identifying the names of
+    the features either directly from the data or from user supplied values. It
+    also provides other functionality for feature selection, e.g. ensuring that
+    a subset of features is used if specified by the user.
 
     Parameters
     ----------
-
-    ax: matplotlib Axes, default: None
+    ax : matplotlib.Axes, default: None
         The axis to plot the figure on. If None is passed in the current axes
         will be used (or generated if required).
 
-    features: list, default: None
-        a list of feature names to use
-        If a DataFrame is passed to fit and features is None, feature
-        names are selected as the columns of the DataFrame.
+    features : list, default: None
+        The names of the features specified by the columns of the input dataset.
+        This length of this list must match the number of columns in X, otherwise
+        an exception will be raised on ``fit()``.
 
-    kwargs : dict
+    kwargs : dict, optional
         Keyword arguments that are passed to the base class and may influence
         the visualization as defined in other Visualizers.
 
+    Attributes
+    ----------
+    features_ : ndarray, shape (n_features,)
+        The names of the features discovered or used in the visualizer that
+        can be used as an index to access or modify data in X. If a user passes
+        feature names in, those features are used. Otherwise the columns of a
+        DataFrame are used or just simply the indices of the data array.
     """
 
     def __init__(self, ax=None, features=None, **kwargs):
         super(MultiFeatureVisualizer, self).__init__(ax=ax, **kwargs)
 
         # Data Parameters
-        self.features_ = features
+        self.features = features
```

Comment:
This change is important - we're saving the hyperparameter independent of the learned parameter so users can compare what is passed into the class and what is "learned" particularly if one is a different type from the other. The learned parameter is always a numpy array. 

Note there is similar new behavior for `classes_` as well.

Reply:
Thanks for clarifying this in the docstring by noting both the parameter `features` and the attribute `features_`!